tests/Performance/Limiter/MemoryLimiterTest.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\Performance\Limiter;
+
+use Infection\Performance\Limiter\MemoryLimiter;
+use Infection\TestFramework\AbstractTestFrameworkAdapter;
+use Memory_Aware\FakeAwareAdapter;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Process\Process;
+
+/**
+ * @internal
+ */
+final class MemoryLimiterTest extends TestCase
+{
+    protected function setUp(): void
+    {
+        if (ini_get('memory_limit') !== '-1') {
+            $this->markTestSkipped('Unable to test if a memory limit is already set.');
+        }
+
+        if (\PHP_SAPI === 'phpdbg' || \extension_loaded('xdebug')) {
+            $this->markTestSkipped('Unable to run tests if PHPDBG or xdebug is active.');
+        }
+    }
+
+    public function test_it_does_nothing_when_adapter_is_not_memory_limit_aware(): void
+    {
+        $fs = $this->createMock(Filesystem::class);
+        $fs->expects($this->never())->method($this->anything());
+
+        $process = $this->createMock(Process::class);
+        $process->expects($this->never())->method($this->anything());
+
+        $adapter = $this->createMock(AbstractTestFrameworkAdapter::class);
+        $adapter->expects($this->never())->method($this->anything());
+
+        $memoryLimiter = new MemoryLimiter($fs, 'foo/bar');
+        $memoryLimiter->applyMemoryLimitFromProcess($process, $adapter);
+    }
+
+    public function test_it_applies_memory_limit_if_possible(): void
+    {
+        $filename = sys_get_temp_dir() . '/fake-ini' . microtime() . '.ini';
```

Comment:
Could we please prepare a workspace for files which we will use as part of this test. For example `infection-test` or whatever else. And on `tearDown` we will need to drop that folder.

an example you can find in the MutationConfigBuilderTest.

Why it's should be done?
Because our users (and me) may mount temporary folder on RAM :)